components/messaging/CurrentConversation.tsx

```diff
@@ -82,58 +84,61 @@ const CurrentConversation: React.FC = () => {
   }, [currentMessages, setCurrentMessages]);
 
   useEffect(() => {
+    let debounce: NodeJS.Timeout;
```

Comment:
I was thinking that by making debounce mutable, I would clear the previous timeout before setting a new one. That way, if there’s more scrolling within the 3 seconds, I could stop the previous timeout and restart it, avoiding unnecessary calls.

But I trust your decision about on it, because I was doing it late at night.

Reply:
The reason is explained and useRef is implemented to avoid mutability